docs/study-def-tricks.md

```diff
@@ -0,0 +1,318 @@
+This section introduces some tricks for working more efficiently with the study definition API, by helping you to avoid copy-and-pasting chunks of code unnecessarily. 
+
+## Create a variable for each code in a codelist
+
+Say we have a codelist and want to return the number of times each code in that codelist was used, then a variable is needed for each code within the codelist. The standard way to create these variables would be as follows:
+
+```py
+study = StudyDefinition(
+
+    ...
+
+    count_code1 = patients.with_these_clinical_events(
+        codelist(code1, system="snomed"),
+        on_or_after="index_date",
+        returning = "number_of_matches_in_period",
+        return_expectations={
+            "incidence": 0.1,
+            "int": {"distribution": "normal", "mean": 3, "stddev": 1}
+        }
+    ),
+
+    count_code2 = patients.with_these_clinical_events(
+        codelist(code2, system="snomed"),
+        on_or_after="index_date",
+        returning = "number_of_matches_in_period",
+        return_expectations={
+            "incidence": 0.1,
+            "int": {"distribution": "normal", "mean": 3, "stddev": 1},
+        }
+    ),
+
+    ...
+	
+)
+
+```
+
+A neater way of doing this is to wrap the variable signature in a function and loop that function over each code in the codelist:
+
+```py
+
+def loop_over_codes(code_list):
+
+    def make_variable(code):
+        return {
+            f"count_{code}": (
+                patients.with_these_clinical_events(
+                    codelist([code], system="snomed"),
+                    on_or_after="index_date",
+                    returning="number_of_matches_in_period",
+                    return_expectations={
+                         "incidence": 0.1,
+                         "int": {"distribution": "normal", "mean": 3, "stddev": 1},
+                    },
+                )
+            )
+        }
+
+    variables = {}
+    for code in code_list:
+        variables.update(make_variable(code))
+    return variables
+
+```
+
+This function is then invoked inside the `StudyDefinition` call for example as follows:
```

Comment:
A bit verbose? "This function can be invoked in the `StudyDefinition` section as follows:"?